src/libasr/asr_utils.h

```diff
@@ -809,6 +809,11 @@ static inline std::string binop_to_str_python(const ASR::binopType t) {
     }
 }
 
+static inline bool is_immutable(const ASR::ttype_t *type) {
+    return ((ASR::is_a<ASR::Character_t>(*type) || ASR::is_a<ASR::Tuple_t>(*type)
+        || ASR::is_a<ASR::Complex_t>(*type)));
+}
```

Comment:
@namannimmo10 This will fail for:
```py
def ff():
    a: c32[2]
    a[0] = complex(1, 2)
    print(a)
    a[1] = complex(3, 4)
    print(a)

ff()
```

Reply:
I think the error should only be raised for,

`a.real = 1.0` or `a.img = 2.0` (i.e., only for updating attributes of complex type). We can always assign a complex variable name to another value. The original value shouldn't have any change. 

If there's an error can you fix this @Smit-create ?